include/mbgl/storage/resource.hpp

```diff
@@ -20,17 +20,42 @@ class Resource {
         SpriteJSON
     };
 
-    Resource(Kind kind_, const std::string& url_)
+    struct TileData {
+        std::string urlTemplate;
+        float pixelRatio;
```

Comment:
Actually, I'm comfortable saying we'll never support non-integer multipliers for tile resolution. I'll make it an integer here and in the database schema.